### PR TITLE
null check cleanup

### PR DESCRIPTION
**Pull request by pq**
A little inelegant but trying to winnow down the `!`s.

/cc @a14n @bwilkerson

### CODE REVIEW

File: lib/src/rules/use_late_for_private_fields_and_variables.dart [hunk 1]
```diff
@@ -177,9 +181,9 @@ class _Visitor extends UnifyingAstVisitor<void> {
 
   void _checkAccess(Iterable<CompilationUnitElement> units) {
     final allNullableAccess =
-        units.expand((unit) => nullableAccess[unit]!).toSet();
+        units.expand((unit) => nullableAccess[unit] ?? {}).toSet();
```
Comment (bwilkerson) [inline lib/src/rules/use_late_for_private_fields_and_variables.dart:184]: Consider making empty collections that are never stored (and hence never added to) be `const`.

File: lib/src/rules/use_late_for_private_fields_and_variables.dart [hunk 2]
```diff
@@ -111,6 +112,9 @@ class _Visitor extends UnifyingAstVisitor<void> {
       element = DartTypeUtilities.getCanonicalElementFromIdentifier(node);
     }
 
+    bool isNonNullable(DartType? type) =>
```
Comment (bwilkerson) [inline lib/src/rules/use_late_for_private_fields_and_variables.dart:115]: Seems like this would be useful to have in the utilities library.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/rules/use_late_for_private_fields_and_variables.dart (hunk 1 above)
Comment (pq) [reply lib/src/rules/use_late_for_private_fields_and_variables.dart:184]: Good catch!

File: lib/src/rules/use_late_for_private_fields_and_variables.dart (hunk 2 above)
Comment (pq) [reply lib/src/rules/use_late_for_private_fields_and_variables.dart]: Good idea!

**Pull request merged**